### PR TITLE
User story 17

### PR DESCRIPTION
**Pull request by DavisC0801**
This PR...
Adds a sorting by time of review functionality to the user show page.
Updates user model to sort reviews by time.
Adds links to the user show page to provide sorting functionality.

### CODE REVIEW

**Comment by chakeresa** [comment]
Resolves #19

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,22 +1,25 @@
 <div id="review-list">
   <header>
     <h2><%= @user.username %></h2>
+    <div id="sort-bar">
+      <a href="<%= user_path(@user) + "?sort=date-desc" %>">Sort by Newest</a>
+      <a href="<%= user_path(@user) + "?sort=date-asc" %>">Sort by Oldest</a>
+    </div>
   </header>
-  <% @user.reviews.each do |review| %>
+  <% @user.sorted_reviews(params[:sort]).each do |review| %>
```
Comment (chakeresa) [inline app/views/users/show.html.erb]: Hm... I don't think we should be using params in the view. Maybe pass in an ivar from the controller: `@sort_order=params[:sort]` and then in the view `@user.sorted_reviews(@sort_order)`. Would that work?

File: spec/features/users/sort_spec.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+      within("#sort-bar") do
+        expect(page).to have_link("Sort by Oldest")
+        expect(page).to have_link("Sort by Newest")
+      end
+
+      expect(current_path).to eq(user_path(@user_1))
+    end
+
+    it "sorts the page by oldest" do
```
Comment (chakeresa) [inline spec/features/users/sort_spec.rb]:
```suggestion
    it "sorts the  user's reviews from oldest to newest" do
```

File: spec/features/users/sort_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      expect(page.all(".review-info")[1]).to have_content("Rating: #{@review_1.rating}/5")
+      expect(page.all(".review-info")[1]).to have_content(@review_1.text)
+      expect(page.all(".review-info")[2]).to have_content(@review_3.title)
+      expect(page.all(".review-info")[2]).to have_content("Rating: #{@review_3.rating}/5")
+      expect(page.all(".review-info")[2]).to have_content(@review_3.text)
+
+      expect(current_path).to eq(user_path(@user_1))
+    end
+
+    it "sorts the page by newest" do
```
Comment (chakeresa) [inline spec/features/users/sort_spec.rb]:
```suggestion
    it "sorts the user's reviews from newest to oldest" do
```

File: app/views/users/show.html.erb (hunk 1 above)
Comment (DavisC0801) [reply app/views/users/show.html.erb]: Refactoring on branch to match MVC conventions.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -23,5 +24,17 @@
...
+    it "sorts reviews by date" do
+      travel 1.day
+      review_1 = @book_1.reviews.create!(title: "This book rocks!", rating: 5, text: "Read it!", user: @user_1)
+      travel_back
+      review_2 = @book_2.reviews.create!(title: "This book sucks!", rating: 1, text: "Don't read it!", user: @user_1)
+      travel 2.day
+      review_3 = @book_3.reviews.create!(title: "This book is OK.", rating: 3, text: "Read it if you want.", user: @user_1)
+
+      expect(@user_1.sorted_reviews("date-desc").to_a).to eq([review_3, review_1, review_2])
+      expect(@user_1.sorted_reviews("date-asc").to_a).to eq([review_2, review_1, review_3])
```
Comment (chakeresa) [inline spec/models/user_spec.rb:37]: We should add a third test here for a non-supported sort_order argument.

Comment (DavisC0801) [reply spec/models/user_spec.rb:37]: Agreed, adding test on branch.

**Pull request merged**